lib/components/qr_tile.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:qrdemo/providers/firestore_provider.dart';
+import 'package:qrdemo/styles/textstyles.dart';
+
+class QRTile extends StatelessWidget {
+  const QRTile({Key? key, this.code, this.time}) : super(key: key);
+
+  final String? code;
+  final Timestamp? time;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text('$code', style: subheading),
+        Text(
+            Provider.of<FirestoreProvider>(context, listen: false)
+                .formatDate(time!),
```

Comment:
So here I have this formateDate function. I used this because in Firebase when you put a DateTime object into its database, it counts it as a timestamp, so the reason I used this function was to format the timestamp back to DateTime object. When the user confirms the scan when the camera reads the QR, it sends the string code along with DateTime. formatting there would be useless since it will turn to timestamp anyway. So once opening the history screen, it'll iterate through the snapshots created from the StreamBuilder. We use that data to then pass it into the QRTile. QR tile has two parameters, code and time. time is formmatted when it goes to didChangeDepencies which is ran before the build method so it can be passed easily as a string when putting into text widget.